for-research/public-likelihoods/public-likelihoods.ipynb

```diff
@@ -0,0 +1,936 @@
+      "channelCat=reg_4l_FF_CR_mu_hf : rob= obs_x_reg_4l_FF_CR_mu_hf  , n_globs= 213\n",
+      "  bin 0 : content= 760.6084269489845 +/- 60.80981724721013  obsData= 786.0\n",
+      "     model_reg_4l_FF_CR_mu_hf_WZb_shapes : 0.03193221875699237 +/- 0.019269764188575726\n",
+      "     model_reg_4l_FF_CR_mu_hf_WZc_shapes : 0.04699630262621213 +/- 0.0209730955634281\n",
+      "     model_reg_4l_FF_CR_mu_hf_WZl_shapes : 0.04341100687452126 +/- 0.018265618195594538\n",
+      "     model_reg_4l_FF_CR_mu_hf_ZZb_shapes : 0.0026313272846891778 +/- 0.0024741896486190303\n",
+      "     model_reg_4l_FF_CR_mu_hf_ZZc_shapes : 0.002863889400032349 +/- 0.0023306801225602886\n",
+      "     model_reg_4l_FF_CR_mu_hf_ZZl_shapes : 0.0012722844840027392 +/- 0.0008523260922273553\n",
+      "     model_reg_4l_FF_CR_mu_hf_fakes_e_HF_shapes : 0.0009599608019925654 +/- 0.0018586862376011074\n",
+      "     model_reg_4l_FF_CR_mu_hf_fakes_e_other_shapes : 0.03350430536738713 +/- 0.0029793978151340175\n",
+      "     model_reg_4l_FF_CR_mu_hf_fakes_mu_HF_shapes : 723.3249142568677 +/- 56.8594336050965\n",
+      "     model_reg_4l_FF_CR_mu_hf_fakes_other_shapes : 31.82899179595303 +/- 16.022119960542522\n",
+      "     model_reg_4l_FF_CR_mu_hf_others_shapes : 2.173927882400492 +/- 1.0907981957035433\n",
+      "     model_reg_4l_FF_CR_mu_hf_tWZ_shapes : 0.08081303972171128 +/- 0.028345488188923018\n",
+      "     model_reg_4l_FF_CR_mu_hf_tZ_shapes : 0.02028417895928669 +/- 0.008647495352707518\n",
+      "     model_reg_4l_FF_CR_mu_hf_ttH_shapes : 2.2956286287191565 +/- 0.20988310431877333\n",
+      "     model_reg_4l_FF_CR_mu_hf_ttZ_shapes : 0.7202958707673872 +/- 0.017631100186744665\n",
+      "[#1] INFO:Eval -- RooRealVar::setRange(obs_x_reg_3l_ttZ) new range named 'coordRange' created with bounds [0,0.73]\n"
+     ]
+    }
+   ],
+   "source": [
+    "for chan in w[\"models\"][\"simPdf\"].bins(): # the 'bins' method will return the channels of a pdf if it is a multi-channel pdf (which simPdf is)\n",
+    "    print(chan.GetName(),\": rob=\",chan.robs()[0].GetName(), \" , n_globs=\", chan.globs().size()) #number of globs typically indicates number of systematics\n",
+    "    for i,bin in enumerate(chan.bins()):\n",
+    "        print(\"  bin\",i,\": content=\",bin.GetContent(),\"+/-\",bin.GetError(),\" obsData=\",bin.GetData(\"obsData\"))\n",
+    "        if i==0: # we print the 'samples' (components) of just the first bin, to save on overprinting\n",
+    "            for samp in chan[\"samples\"].components():\n",
+    "                print(\"    \",samp.GetName(),\":\",samp.bins()[0].GetContent(),\"+/-\",samp.bins()[0].GetError())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "78680adb-32c2-450c-9310-9945b6fb7cdd",
+   "metadata": {},
+   "source": [
+    "## 1.3. Visualizing the workspace\n",
+    "\n",
+    "The whole model can also be visualized using the following methods:"
```

Comment:
Can you explain a little better the contents of the plot once it's drawn?

Reply:
Hmm, what do you mean? Just add a comment that each channel is shown as an individual plot? Not sure how each individual plot can be explained better...

Reply:
What's the x-axis? What's the y-axis? What are the lines? Any features someone should identify?

Reply:
The y-axis are all labelled automatically (Events). The x-axis labels are taken from the workspace, so the json workspace should be modified to include sensible titles for each of the observables in the channels in my opinion. Likewise, the titles of the channels come from the workspace itself, so the json should be improved. Which are the lines you are referring to? To be able to comment on any specific features of this analysis would need someone from the analysis, I think. Just in case you are reviewing this notebook blind, here's what you get as the output of this cell:
![Screenshot 2024-05-03 at 17 45 03](https://github.com/atlas-outreach-data-tools/notebooks-collection-opendata/assets/18280829/6a303f98-c71f-431f-b4da-d3b909ce0d28)

Reply:
Ok, then to be clear, here is what I'd write:

Each panel of this plot shows the number of events in one channel (usually an analysis region, like a control region or a signal region) in the workspace. Some have several bins (like those in the top row), and some only have a single bin (two of the channels in the bottom row). In each panel the components of the background are drawn in different colors; a single background might enter in one or more channels. A statistical uncertainty is shown on the observed data to guide the eye; the systematic and statistical uncertainties, all captured as parameters in the workspace, are shown as a hatched band on the total background prediction.

Does that make the idea more clear?